src/Products/PythonScripts/PythonScript.py

```diff
@@ -61,6 +61,12 @@
     Python_magic = imp.get_magic()
     del imp
 
+
+# Following identifiers are not allowed to use when instantiating
+# PythonScript due to interaction with RestrictedPython.
+BAD_IDENTIFIERS = [
+    'context', 'container', 'script', 'traverse_subpath',
+]
```

Comment:
Do not use an explicit list to define the problematic names, use instead the same logic as in `PythonScript._compile`: the problematic names are the `bind_names` computed at the beginning of `_compile`.

Reply:
Thank you for pointing me to the right spot!

I did not make "bind_names" an instance variable, as it is just generated twice - or would you prefer otherwise?